subsys/net/l2/Kconfig

```diff
@@ -72,6 +72,20 @@ source "subsys/net/l2/ppp/Kconfig"
 
 source "subsys/net/l2/ieee802154/Kconfig"
 
+config IEEE802154_NET_IF_NO_AUTO_START
```

Comment:
you could move that into subsys/net/l2/ieee802154/Kconfig before the menu, it would be a little bit more logical than here in the generic L2 side.

Reply:
it applies to subsys/net/l2/ieee802154/ and subsys/net/l2/openthread/. but can't put it into both.

Reply:
yes, it's messy. 
We could put it into drivers/ieee802154/Kconfig (note the name of the option does not own _L2_ in it ),  though I do agree that handling the option should be in L2.

Btw, could you also provide a patch that gets rid of IEEE802154_RF2XX_NET_IF_NO_AUTO_START in this same PR? since this more generic option will nullify that older one.

Reply:
> Btw, could you also provide a patch that gets rid of IEEE802154_RF2XX_NET_IF_NO_AUTO_START in this same PR? since this more generic option will nullify that older one.

That's already part of this patch. Did I miss it a certain file?

Reply:
oh yes, nevermind

Reply:
>We could put it into drivers/ieee802154/Kconfig (note the name of the option does not own L2 in it ), ...

Moved it there.